lib/clean/migrate_specialist_subscriber_lists.rb

```diff
@@ -0,0 +1,88 @@
+module Clean
+  # A document_type filter was incorrectly added to some subscriber lists
+  # via a change to specialist publisher.
+  class MigrateSpecialistSubscriberLists
+    def migrate_subscribers_to_working_lists(dry_run: true)
+      lists.each { |from_list|
+        to_list = find_to_list(from_list)
+        migrate_list(from_list, to_list, dry_run: dry_run)
+      }
+
+      puts "Found #{dry_run ? '' : 'and migrated '}#{lists.count} #{'lists'.pluralize(lists.count)}"
+    end
+
+    def lists
+      @lists ||= begin
+        lists = SubscriberList.where(created_at: Date.new(2019, 12, 13)..Date.new(2020, 1, 22))
+        lists.select { |l|
+          l.tags != {} && l.title.exclude?("Brexit") &&
+            l.matched_content_changes.count.zero? &&
+            l.subscribers.activated.count.positive? && l.tags.key?(:document_type)
+        }
+      end
+    end
+
+  private
+
+    def find_to_list(from_list)
+      new_tags = from_list.tags
+      new_tags.delete :document_type
+      query = FindExactQuery.new(
+        tags: new_tags,
+        links: from_list.links,
+        document_type: from_list.document_type,
```

Comment:
Should this be searching on `format`?

Reply:
In this case no, since `document_type` is an attribute of a SubscriberList, rather than part of the `tags`. The bug that occurred is that `document_type` was added to the SubscriberList tags.